src/contracts/misc/GhoStewardV2.sol

```diff
@@ -0,0 +1,286 @@
+
+  /// @inheritdoc IGhoStewardV2
+  uint256 public constant MINIMUM_DELAY = 7 days;
+
+  /// @inheritdoc IGhoStewardV2
+  address public immutable POOL_ADDRESSES_PROVIDER;
+
+  /// @inheritdoc IGhoStewardV2
+  address public immutable GHO_TOKEN;
+
+  /// @inheritdoc IGhoStewardV2
+  address public immutable RISK_COUNCIL;
+
+  GhoDebounce internal _ghoTimelocks;
+  mapping(address => uint40) _facilitatorsBucketCapacityTimelocks;
+  mapping(address => GsmDebounce) internal _gsmTimelocksByAddress;
+
+  mapping(address => bool) internal _controlledFacilitatorsByAddress;
+  EnumerableSet.AddressSet internal _controlledFacilitators;
+
+  mapping(uint256 => address) internal _ghoBorrowRateStrategiesByRate;
+  EnumerableSet.AddressSet internal _ghoBorrowRateStrategies;
+  mapping(uint256 => mapping(uint256 => address)) internal _gsmFeeStrategiesByRates;
+  EnumerableSet.AddressSet internal _gsmFeeStrategies;
+
+  /**
+   * @dev Only Risk Council can call functions marked by this modifier.
+   */
+  modifier onlyRiskCouncil() {
+    require(RISK_COUNCIL == msg.sender, 'INVALID_CALLER');
+    _;
+  }
+
+  /**
+   * @dev Only methods that are not timelocked can be called if marked by this modifier.
+   */
+  modifier notTimelocked(uint40 timelock) {
+    require(block.timestamp - timelock > MINIMUM_DELAY, 'DEBOUNCE_NOT_RESPECTED');
+    _;
+  }
+
+  /**
+   * @dev Constructor
+   * @param owner The address of the owner of the contract
+   * @param addressesProvider The address of the PoolAddressesProvider of Aave V3 Ethereum Pool
+   * @param ghoToken The address of the GhoToken
+   * @param riskCouncil The address of the risk council
+   */
+  constructor(address owner, address addressesProvider, address ghoToken, address riskCouncil) {
+    require(owner != address(0), 'INVALID_OWNER');
+    require(addressesProvider != address(0), 'INVALID_ADDRESSES_PROVIDER');
+    require(ghoToken != address(0), 'INVALID_GHO_TOKEN');
+    require(riskCouncil != address(0), 'INVALID_RISK_COUNCIL');
+
+    POOL_ADDRESSES_PROVIDER = addressesProvider;
+    GHO_TOKEN = ghoToken;
+    RISK_COUNCIL = riskCouncil;
+
+    _transferOwnership(owner);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateFacilitatorBucketCapacity(
+    address facilitator,
+    uint128 newBucketCapacity
+  ) external onlyRiskCouncil notTimelocked(_facilitatorsBucketCapacityTimelocks[facilitator]) {
+    require(_controlledFacilitatorsByAddress[facilitator], 'FACILITATOR_NOT_CONTROLLED');
+    (uint256 currentBucketCapacity, ) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(facilitator);
+    require(
+      _isIncreaseLowerThanMax(currentBucketCapacity, newBucketCapacity, currentBucketCapacity),
+      'INVALID_BUCKET_CAPACITY_UPDATE'
+    );
+
+    _facilitatorsBucketCapacityTimelocks[facilitator] = uint40(block.timestamp);
+
+    IGhoToken(GHO_TOKEN).setFacilitatorBucketCapacity(facilitator, newBucketCapacity);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGhoBorrowCap(
```

Comment:
does it make sense to allow adjusting borrow cap, given the risk steward already can?
So there's two contracts with different timelocks and limitations.

Reply:
We wanted to give this steward management of all GHO parameters so i think it's okay since both stewards have different scopes.
This comes in handy because the borrow cap for GHO relies on facilitators bucket cap, which is also handed here.

Reply:
Yes, GHO stewards has FInance & Growth SPs on it, while capPlussteward has only risk in it, not same scope and should have different limitations

Reply:
Please @marczeller clarify this configuration on the governance forum thread.

Reply:
https://governance.aave.com/t/arfc-gho-stewards/16466/7 done